### ISSUE TITLE
Incorporate CLP "Action Links" for distinguish buttons and links

### ISSUE THREAD

**Issue by brianalloyd** [department-of-veterans-affairs/va.gov-team#21929]
## Issue Description
_We need to update bottons and links on CLP with the new "Action Links" design components for MVP 1.0. The action link is an eye-catching link to start a digital service. An action link entices users to take action. Example: Starting a benefit application_

Design Link is [HERE](https://design.va.gov/components/action-links) 

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Validate action links are rendering properly 
- [ ] Ensure action links are tested and validated in pre-prod environment
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] The data returned matches API response (for given user or scenario)
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by brianalloyd** [comment]
@rtwell big ask, could you provide an updated CLP mock up reflecting the incorporation of the "Action Link" placements as were a little unclear in terms of what element to use with which CTA/Link.  Once we have an updated design mockup we'll scrub into updating FE to reflect design elements in support.  Note: We would love to implement these changes prior to launch if possible.  cc @johnhashva

**Comment by rtwell** [comment]
no prob—i can get this to you tomorrow 3/24/21

**Comment by rtwell** [comment]
Here be the comp ([also inspectable here](https://www.sketch.com/s/b0ab7a47-11d3-4175-90c1-c54638ca9925/v/45pznD/a/09pKVEj/r/5oM8qE#Inspector))! One thing to note about the new action links—we will need a reverse variant for the purposes of this page. The design system team will make it available asap, or we can create the variant and give them to code, whichever you prefer. Either way, you can see the [spec for it here](https://www.sketch.com/s/fa5c1448-a794-4d67-aca9-661ec628a807/a/paQqelG#Inspector). 

![image](https://user-images.githubusercontent.com/12564977/112497634-2f007980-8d5c-11eb-8378-e476febc1697.png)

**Comment by brianalloyd** [comment]
Thanks @rtwell for the design comps, very helpful.  @kelsonic now that we have the design specs going to move this up in order of priority for this sprint and can discuss the details in SU.  I'm also going to add @rtwell on the AC for this ticket to ensure all design and styling is rendering correctly in CMS preview once changes are applied.

**Comment by kelsonic** [comment]
PR merged + will be ready for validation tomorrow following daily deploy 👍

**Comment by brianalloyd** [comment]
@kelsonic this looks great.  @johnhashva and @rtwell can you both take a look at the updated action links and validate on your end?  TIA.

**Comment by johnhashva** [comment]
@brianalloyd + @kelsonic -- Looks super to me -- ty!

**Comment by rtwell** [comment]
forgive my ignorance, but where might i see this in staging (or w/e environment)

**Comment by johnhashva** [comment]
@rtwell here you go:

https://staging.va.gov/campaign-mission-act/
PW: clptest123

**Comment by rtwell** [comment]
perfect, tyvm john

1. [actions links](https://design.va.gov/components/action-links) should not have a `>` at the end. the first couple instances don't have them, but the rest do. example:
![image](https://user-images.githubusercontent.com/12564977/114450956-aaf42180-9ba4-11eb-9492-909fffb2bd84.png)

2. it looks like there is a space character that is getting underlined before the first letter in the link. is that baked into how it's built in the design system or limited to the CLP instances of it? example:
![image](https://user-images.githubusercontent.com/12564977/114451202-f9a1bb80-9ba4-11eb-9ab2-0069bba873f8.png)

**Comment by brianalloyd** [comment]
Thanks @rtwell for the feedback and review.  @kelsonic when your back in the saddle tomorrow, let's prioritize this update.  thanks.

**Comment by kelsonic** [comment]
#1 I can knock out right now. #2 is a design system issue 🙂

**Comment by kelsonic** [comment]
PR is up for #1 ^ 🙂

**Comment by kelsonic** [comment]
PR merged

**Comment by bkjohnson** [comment]
@kelsonic @rtwell I don't think that underline issue is with the design system. The link text appears to have additional whitespace arround it:

![image](https://user-images.githubusercontent.com/2008881/114746023-202a3880-9d04-11eb-97cb-9d15fce92483.png)

When I remove this whitespace, the underlined whitespace also gets removed:

![image](https://user-images.githubusercontent.com/2008881/114746308-71d2c300-9d04-11eb-9cdb-79e710afab73.png)

I think this extra whitespace is either coming from the template or the content.

**Comment by kelsonic** [comment]
Ah ha! Good catch! It's the content, will get a PR up here shortly for it, thanks @bkjohnson! 🙂

**Comment by brianalloyd** [comment]
Thanks @kelsonic and @bkjohnson for catching this and updating accordingly.

**Comment by brianalloyd** [comment]
Validated and closing.
